src/app/AppContainer.js

```diff
@@ -94,95 +95,77 @@ class AppContainer extends React.Component<PropsType> {
 			this.toggleMode();
 		} );
 
-		this.subscriptionParentSetTitle = subscribeSetTitle( ( payload ) => {
-			this.props.editTitle( payload.title );
-		} );
-
 		this.subscriptionParentUpdateHtml = subscribeUpdateHtml( ( payload ) => {
 			this.updateHtmlAction( payload.html );
 		} );
+
+		this.subscriptionParentSetTitle = subscribeSetTitle( ( payload ) => {
+			this.props.editTitle( payload.title );
+		} );
 	}
 
 	componentWillUnmount() {
 		if ( this.subscriptionParentGetHtml ) {
 			this.subscriptionParentGetHtml.remove();
 		}
+
 		if ( this.subscriptionParentToggleHTMLMode ) {
 			this.subscriptionParentToggleHTMLMode.remove();
 		}
+
+		if ( this.subscriptionParentUpdateHtml ) {
+			this.subscriptionParentUpdateHtml.remove();
+		}
+
 		if ( this.subscriptionParentSetTitle ) {
 			this.subscriptionParentSetTitle.remove();
 		}
-		if ( this.subscriptionParentUpdateHtml ) {
-			this.subscriptionParentUpdateHtml.remove();
+	}
+
+	componentDidUpdate( prevProps ) {
+		if ( ! prevProps.isReady && this.props.isReady ) {
+			const blocks = this.props.getEditorBlocks();
+			const isUnsupportedBlock = ( { name } ) => name === getUnregisteredTypeHandlerName();
+			const unsupportedBlocks = blocks.filter( isUnsupportedBlock );
+			const hasUnsupportedBlocks = ! isEmpty( unsupportedBlocks );
+
+			RNReactNativeGutenbergBridge.editorDidMount( hasUnsupportedBlocks );
 		}
 	}
 
+	toggleMode() {
+		const { mode, switchMode } = this.props;
+		switchMode( mode === 'visual' ? 'text' : 'visual' );
+	}
+
 	serializeToNativeAction() {
 		if ( this.props.mode === 'text' ) {
 			this.updateHtmlAction( this.props.getEditedPostContent() );
 		}
 
-		const html = serialize( this.props.blocks );
-		const title = this.props.title;
+		const html = serialize( this.props.getEditorBlocks() );
+		const title = this.props.getEditedPostAttribute( 'title' );
 
-		const hasChanges = title !== this.lastTitle || html !== this.lastHtml;
+		const hasChanges = title !== this.post.title.raw || html !== this.post.content.raw;
```

Comment:
I wonder, is the serialization/parsing pair symmetric nowadays? I remember we needed the `this.lastHtml = serialize( parse( props.initialHtml || '' ) )` dance because the wasn't symmetric. Have you perhaps tried backing out of the block editor in WPAndroid and the post is not marked as changes if no user edits occurred?

Reply:
I haven't tried but I can have a look with our sample post (`initial-html.js`)